src/App.tsx

```diff
@@ -50,15 +48,21 @@
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector value={author} onChange={setAuthor} />
+                <UserSelector
+                  users={users}
+                  value={selectedUser}
+                  onChange={setSelectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                {!author && <p data-cy="NoSelectedUser">No user selected</p>}
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
 
-                {author && !loaded && <Loader />}
+                {selectedUser && !postsLoaded && <Loader />}
```

Comment:
The condition `selectedUser && !postsLoaded` correctly shows the loader while posts are being fetched. Ensure that `postsLoaded` is accurately reflecting the loading state.